core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1043,6 +1043,17 @@ public PluginWrapper getInstalled() {
             return pm.getPlugin(name);
         }
 
+        @Exported
+        public boolean isCompatible() {
```

Comment:
Would `@Restricted(NoExternalUse.class)` this, no reason IMO to expose this to plugins.

If we end up not doing that, needs Javadoc + `@since TODO`.

Reply:
For this case I've included the Javadoc and the `@since TODO`.

I might be wrong, but I see this new method as the replacement of the one that was being called previously (`isCompatibleWithInstalledVersion`), which is being exported; so I thought it could be useful to export this one so that whoever uses the "old" one can switch to this new and more comprehensive check

Reply:
Since nobody else has commented on this, I've applied @daniel-beck 's suggestion and made the `isCompatible` methid restricted (`@Restricted(NoExternalUse.class)`).

Definitely not necessary to expose it to plugins for now. And if in the future someone sees value in calling it from a plugin this can always be changed.